pkg/kubelet/dockershim/security_context.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package dockershim
+
+import (
+	"fmt"
+	"strconv"
+
+	dockercontainer "github.com/docker/engine-api/types/container"
+	runtimeapi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+const (
+	dockerLabelUser  string = "label:user"
+	dockerLabelRole  string = "label:role"
+	dockerLabelType  string = "label:type"
+	dockerLabelLevel string = "label:level"
+)
+
+// applySandboxSecurityContext updates docker sandbox options according to security context.
+func applySandboxSecurityContext(lc *runtimeapi.LinuxPodSandboxConfig, config *dockercontainer.Config, hc *dockercontainer.HostConfig) {
+	if lc == nil {
+		return
+	}
+
+	modifySandboxContainerConfig(lc.SecurityContext, config)
+	modifySandboxHostConfig(lc.SecurityContext, hc)
+}
+
+// applyContainerSecurityContext updates docker container options according to security context.
+func applyContainerSecurityContext(lc *runtimeapi.LinuxContainerConfig, sandboxID string, config *dockercontainer.Config, hc *dockercontainer.HostConfig) {
+	if lc == nil {
+		return
+	}
+
+	modifyContainerConfig(lc.SecurityContext, config)
+	modifyHostConfig(lc.SecurityContext, sandboxID, hc)
+	return
+}
+
+// modifySandboxContainerConfig applies sandbox security context config to dockercontainer.Config.
+func modifySandboxContainerConfig(podSc *runtimeapi.SandboxSecurityContext, config *dockercontainer.Config) {
+	if podSc != nil && podSc.RunAsUser != nil {
+		modifyContainerConfig(&runtimeapi.SecurityContext{
+			RunAsUser: podSc.RunAsUser,
+		}, config)
```

Comment:
Here casts `SandboxSecurityContext` to `SecurityContext`. It could reuse `modifyContainerConfig` acutally. Will update.